src/component/solenoid.go

```diff
@@ -8,12 +8,65 @@ import (
 	"time"
 )
 
-// Solenoid - base component + solenoid specific metadata
+// Solenoid - SolenoidConfig + GPIO
 type Solenoid struct {
-	BaseComponent `yaml:",inline"`
+	GPIO           io.Gpio
+	SolenoidConfig `yaml:",inline"`
+}
+
+//SolenoidConfig - Static config values for transport and storage
+type SolenoidConfig struct {
 	Type          SolenoidType `yaml:"type"`
 	Mode          SolenoidMode `yaml:"mode"`
-	GPIO          io.Gpio
+	BaseComponent `yaml:",inline"`
+}
+
+// SolenoidType -
+type SolenoidType string
+
+const (
+	// NormallyClosed represents a solenoid that does not allow flow without power
+	NormallyClosed SolenoidType = "NC"
+	// NormallyOpen represents a solenoid that is allows flow without power
+	NormallyOpen = "NO"
+)
+
+//SolenoidMode -
+type SolenoidMode string
+
+const (
+	//Supply - tank supply, pilot supply and transport solenoids
+	Supply SolenoidMode = "supply"
+	//Outlet - propane exhaust solenoid
+	Outlet = "outlet"
+)
+
+//DoNotCloseDuration - placeholder
+const DoNotCloseDuration = -1
+
+//GetConfig -
+func (s Solenoid) GetConfig() SolenoidConfig {
+	config := SolenoidConfig{
+		Type: s.Type,
+		Mode: s.Mode,
+	}
+	config.UID = s.UID
```

Comment:
this set of maps along with the ones starting on line 63 are very similar which leads me to believe that it should be encapsulated into a function that will map values from one object to another if they exist

Reply:
Isn't that what GetConfig and Load() are? Are you asking me to verify that the key is in the struct before loading it into the Solenoid obj?

Reply:
I mean this could be converted into a loop through a config array. so the config array would be something like `[ 'UID', 'Enabled', 'Name' ]` and then you loop through that and apply the source to the dest. Because some data is got and more of it is loaded that array may have to have some flags on it like onLoad = true or false. This makes the mapping done in a configurable array rather than a series of declarations.

But also whats being done here works fine and if you want to leave it its fine

Reply:
I'll try and make it something more generic in the future.